tt_metal/impl/dispatch/arch.cpp

```diff
@@ -0,0 +1,738 @@
+    {127, 8, 0, MUX_D,              {132, 133,   x,   x}, {126,   x,   x,   x}, NOC::NOC_0, NOC::NOC_0, NOC::NOC_0},
+    {128, 8, 0, PACKET_ROUTER_DEMUX,{126,   x,   x,   x}, {130,   x,   x,   x}, NOC::NOC_0, NOC::NOC_0, NOC::NOC_0},
+    {129, 8, 0, PACKET_ROUTER_DEMUX,{126,   x,   x,   x}, {131,   x,   x,   x}, NOC::NOC_0, NOC::NOC_0, NOC::NOC_0},
+    {130, 8, 0, PREFETCH_D,         {128,   x,   x,   x}, {132, 134,   x,   x}, NOC::NOC_0, NOC::NOC_0, NOC::NOC_0},
+    {131, 8, 1, PREFETCH_D,         {129,   x,   x,   x}, {133, 135,   x,   x}, NOC::NOC_0, NOC::NOC_0, NOC::NOC_0},
+    {132, 8, 0, DISPATCH_D,         {130,   x,   x,   x}, {134, 127,   x,   x}, NOC::NOC_0, NOC::NOC_1, NOC::NOC_0},
+    {133, 8, 1, DISPATCH_D,         {131,   x,   x,   x}, {135, 127,   x,   x}, NOC::NOC_0, NOC::NOC_1, NOC::NOC_0},
+    {134, 8, 0, DISPATCH_S,         {130,   x,   x,   x}, {132,   x,   x,   x}, NOC::NOC_1, NOC::NOC_1, NOC::NOC_1},
+    {135, 8, 1, DISPATCH_S,         {130,   x,   x,   x}, {132,   x,   x,   x}, NOC::NOC_1, NOC::NOC_1, NOC::NOC_1},
+};
+
+std::vector<FDKernel *> node_id_to_kernel;
+
+// Helper function to get the nodes for this platform
+std::vector<dispatch_kernel_node_t> get_nodes(const std::set<chip_id_t> &device_ids, uint32_t num_hw_cqs) {
+    // Select/generate the right input table, depends on (1) board [detected from total # of devices], and (2) number
+    // of active devices. TODO: read this out of YAML instead of the structs above?
+    uint32_t total_devices = tt::Cluster::instance().number_of_devices();
+    TT_ASSERT(total_devices == 1 or total_devices == 2 or total_devices == 4 or total_devices == 8 or total_devices == 36, "Unexpected target.");
+    uint32_t num_devices = device_ids.size();
+    tt::log_debug("FD Config: {}/{} devices, {} HW CQs", num_devices, total_devices, num_hw_cqs);
+    TT_ASSERT(num_devices > 0, "Can't determine dispatch architecture with no active devices.");
+    TT_ASSERT(num_devices <= total_devices);
+    std::vector<dispatch_kernel_node_t> nodes;
+
+    std::set<chip_id_t> mmio_devices;
+    std::set<chip_id_t> remote_devices;
+    for (auto id : device_ids) {
+        if (tt::Cluster::instance().get_associated_mmio_device(id) == id) {
+            mmio_devices.insert(id);
+        } else {
+            remote_devices.insert(id);
+        }
+    }
+    // Supported grid either has one remote per mmio or none
+    tt::log_debug("FD Config: mmio_count={}, remote_count={}", mmio_devices.size(), remote_devices.size());
+
+    // Helper function to get nodes for single device
+    auto populate_single_device = [&]() {
+        if (num_hw_cqs == 1) {
+            return single_card_arch_1cq;
+        } else {
+            // Special case here, single-device can either have dispatch_s or no dispatch_s, depending on the dispatch
+            // core type. This is only an issue for single-chip, since multi-chip always has ethernet dispatch (and
```

Comment:
I don't think this is true we could be running tensix dispatch for multi-chip as well